src/WebJobs.Script.WebHost/TokenExpiration/TokenExpirationService.cs

```diff
@@ -0,0 +1,184 @@
+            };
+
+            // We only want to parse AzureWebJobsStorage in the following format
+            // BlobEndpoint=<blob-endpoint-url>;QueueEndpoint=<queue-endpoint-url>;TableEndpoint=<table-endpoint-url>;SharedAccessSignature=<sas-token>
+            var azureWebJobsStorage = _environment.GetEnvironmentVariable(EnvironmentSettingNames.AzureWebJobsSecretStorage);
+
+            // Check if the value starts with "BlobEndpoint=" and contains "SharedAccessSignature="
+            if (azureWebJobsStorage.StartsWith("BlobEndpoint=", StringComparison.OrdinalIgnoreCase)
+                && azureWebJobsStorage.Contains("SharedAccessSignature="))
+            {
+                appSettings.Add(EnvironmentSettingNames.AzureWebJobsSecretStorage, azureWebJobsStorage);
+            }
+
+            EmitLogging(appSettings);
+        }
+
+        private void EmitLogging(Dictionary<string, string> appSettings)
+        {
+            foreach (var setting in appSettings)
+            {
+                var uri = setting.Value;
+                if (!string.IsNullOrEmpty(uri))
+                {
+                    string sasTokenExpirationDate;
+
+                    if (setting.Key == EnvironmentSettingNames.AzureWebJobsSecretStorage)
+                    {
+                        sasTokenExpirationDate = Utility.GetSasTokenExpirationDateFromSasSignature(uri);
+                    }
+                    else
+                    {
+                        sasTokenExpirationDate = Utility.GetSasTokenExpirationDate(new Uri(uri));
+                    }
+
+                    if (!string.IsNullOrEmpty(sasTokenExpirationDate))
+                    {
+                        var parsedDate = DateTime.Parse(sasTokenExpirationDate);
+                        var currentDate = DateTime.Now.Date;
+
+                        var difference = parsedDate.Subtract(currentDate);
+
+                        // Log an error event if the token is already expired; otherwise log a warning event
+                        if (difference.TotalDays <= 0)
+                        {
+                            string message = string.Format(Resources.SasTokenExpiredFormat, setting.Key);
+                            DiagnosticEventLoggerExtensions.LogDiagnosticEventError(_logger, message, DiagnosticEventConstants.SasTokenExpiringErrorCode, DiagnosticEventConstants.SasTokenExpiringErrorHelpLink, new Exception(message));
+                        }
+                        if (difference.TotalDays <= 30)
+                        {
+                            string message = string.Format(Resources.SasTokenExpiringFormat, (int)difference.TotalDays, setting.Key);
+                            DiagnosticEventLoggerExtensions.LogDiagnosticEvent(_logger, Microsoft.Extensions.Logging.LogLevel.Warning, 0, DiagnosticEventConstants.SasTokenExpiringErrorCode, message, DiagnosticEventConstants.SasTokenExpiringErrorHelpLink, null);
```

Comment:
nit: consider using named argument for the `null` value (last param) to improve readability.